### ISSUE TITLE
The command `antiword' failed because the executable `antiword` is not installed on your system

### ISSUE THREAD

**Issue by amjadparacha** [deanmalmgren/textract#297]
Hi,

I'm facing following error when running textract on Ubuntu 18.04. Flask application is running as a service and behind nginx.

> The command `antiword' failed because the executable `antiword` is not installed on your system. Please make sure the appropriate dependencies are installed before using textract:
> 
> http://textract.readthedocs.org/en/latest/installation.html
> Traceback (most recent call last):
>   File "/home/jenkins/jenkins_qa/workspace/AgileDefense/qa_push_job/flask-app/qa/lib/python3.6/site-packages/textract/parsers/utils.py", line 84, in run
>     stdout=subprocess.PIPE, stderr=subprocess.PIPE,
>   File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
>     restore_signals, start_new_session)
>   File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
>     raise child_exception_type(errno_num, err_msg, err_filename)
> FileNotFoundError: [Errno 2] No such file or directory: 'antiword': 'antiword'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "src/attachment_extraction.py", line 26, in extractTextFromAttachment
>     attachmentText = textract.process(file).decode("utf-8")
>   File "/home/jenkins/jenkins_qa/workspace/AgileDefense/qa_push_job/flask-app/qa/lib/python3.6/site-packages/textract/parsers/__init__.py", line 77, in process
>     return parser.process(filename, encoding, **kwargs)
>   File "/home/jenkins/jenkins_qa/workspace/AgileDefense/qa_push_job/flask-app/qa/lib/python3.6/site-packages/textract/parsers/utils.py", line 46, in process
>     byte_string = self.extract(filename, **kwargs)
>   File "/home/jenkins/jenkins_qa/workspace/AgileDefense/qa_push_job/flask-app/qa/lib/python3.6/site-packages/textract/parsers/doc_parser.py", line 9, in extract
>     stdout, stderr = self.run(['antiword', filename])
>   File "/home/jenkins/jenkins_qa/workspace/AgileDefense/qa_push_job/flask-app/qa/lib/python3.6/site-packages/textract/parsers/utils.py", line 91, in run
>     ' '.join(args), 127, '', '',
> textract.exceptions.
> 

Both of these packages are installed similar to local environment, where it is running without any issue

**Comment by jpweytjens** [comment]
I have no experience with Flask. It seems however that the Flask environment has no acces to the installed antiword binary. Does Flask change the path?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by amjadparacha** [comment]
Not sure about that, how can i check it? If it is doing so why its doing it for 'antiword' specifically because there are other office libraries being used in application and they are working without issue.

**Comment by jpweytjens** [comment]
I'm not sure I can be of much help here. If textract and antiword work in your local environment, the problem seems to be related Flask and the environment in which it is run. It is hard to diagnose the cause with my limited knowledge of Flask and the information I have about your setup.

Given that, I can only think of some basic things. Is Flask run in a different virtual environment?  If so, is antiword available in this environment? You can find where antiword is loated with the following

```
which antiword
```

and then check whether that location is in your `PATH` with

```
echo $PATH
```

If antiword is missing in the Flask environment, install in that environment as well. If that doesn't work, you should provide more information about your setup, specifically your (virtual) environment(s), how you installed textract and how you installed antiword.


<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by arslanather** [comment]
I was able to resolve this problem by adding `enviroment=PATH="/usr/bin"` in my supervisor configuration which i was using to run flask app using gunicorn. You should look if you are overriding `PATH` somewhere in your configurations.

**Comment by amjadparacha** [comment]
I've tried above as well as other things but nothing worked out. Therefore, as a last option I've created a new environment on a brand new server and ran my code, it ran as smooth as it can without any issue. Seems like there was an issue with OS libraries.

Thanks for looking into it, I've learned lots of new things.

PS: I've already executed `sudo apt update` and `sudo apt -y upgrade`, but it didn't fixed the problem.
